examples/standalone/runfile/dynamics.py

```diff
@@ -1,120 +1,152 @@
+import json
+import pathlib
 from argparse import ArgumentParser
+from datetime import datetime
 
-import mpi4py
+import git
+import numpy as np
 import serialbox
 import yaml
+from mpi4py import MPI
 
 import fv3core
 import fv3core._config as spec
 import fv3core.stencils.fv_dynamics as fv_dynamics
 import fv3core.testing
 import fv3gfs.util as util
-from fv3core.testing import GlobalTimer, write_to_json
+from fv3gfs.util import Timer
+
+
+def print_and_write_global_timings(
+    timer, experiment_name, time_step, backend, comm, root=0
+):
+    is_root = comm.Get_rank() == root
+    recvbuf = np.array(0.0)
+    experiment = {}
+    now = datetime.now()
+    sha = git.Repo(
+        pathlib.Path(__file__).parent.absolute(), search_parent_directories=True
+    ).head.object.hexsha
```

Comment:
Also OK with this as-is, but a more extensible/re-usable pattern would be to pass in a "label" for the report from the command-line. Having this low-level code here adds an extra responsibility to the function.

Reply:
Can you add a try: except block around this sha line which puts a default value if this is run not in a git repo?